app/actions/index.js

```diff
@@ -32,11 +37,11 @@ export const createDat = () => dispatch => {
   addDat({ path })(dispatch)
 }
 
-export const addDat = ({ key, path, paused, ...opts }) => dispatch => {
+export const addDat = ({ key, path, ...opts }) => async dispatch => {
   if (key) key = encode(key)
   if (!path) path = `${homedir()}/Downloads/${key}`
+  const unlock = await datsLock(key)
```

Comment:
did you run into any locking issues here before? Instead we could tweak this function to return early if a dat with given key already exists, this would be our "lock"

Reply:
I wrapped it as a safety measure but didn't run into issues yet. Didn't think of returning a dat immediately 😊That might be possible (and better) but there is also the `storeOnDisk` operation way down the road that should be done as well. Which maybe, makes a lock reasonable anyways (think on it?)

Reply:
cam we talk about locks in a separate issue? this is trying to do quite a lot and makes it harder to discuss

Reply:
I am added the lock on data because I added locking but I think the lock on the index should be part of the persistence PR

Reply:
I am added the lock on data because I added locking but I think the lock on the index should be part of the persistence PR

Reply:
I am added the lock on data because I added locking but I think the lock on the index should be part of the persistence PR

Reply:
I am added the lock on data because I added locking but I think the lock on the index should be part of the persistence PR

Reply:
I am added the lock on data because I added locking but I think the lock on the index should be part of the persistence PR